benchmarks/benchmark_sed.py

```diff
@@ -0,0 +1,150 @@
+"""This file contains code that performs benchmarks for the processor workflows
+"""
+import os
+import timeit
+from importlib.util import find_spec
+
+import dask
+import numpy as np
+import psutil
+
+from sed import SedProcessor
+from sed.binning.binning import bin_dataframe
+
+package_dir = os.path.dirname(find_spec("sed").origin)
+
+
+num_cores = min(20, psutil.cpu_count())
+# use fix random numbers for comparability
+np.random.seed(42)
+# 100 Billion events, ~ 3 GByte.
+n_pts = 100000000
+ranges = np.array([[0, 2048], [0, 2048], [60000, 120000], [2000, 20000]])
+axes = ["X", "Y", "t", "ADC"]
+array = (
+    dask.array.random.random((n_pts, len(ranges))) * (ranges[:, 1] - ranges[:, 0]) + ranges[:, 0]
+)
+dataframe = dask.dataframe.from_dask_array(array, columns=axes)
+
+
+target_artificial_1d = 2.68
```

Comment:
I'd propose that these values are stored in a yaml file inside this folder. Everytime the values improve, they are updated in that file. So we always have an up to date comparison and hence pass test if performance gets worse than current state.

Reply:
That is a good idea to automatically update targets If they decrease. I can look into this

Reply:
Now with targets in yaml file, and PR mechanism for updates

Reply:
Looks good.

From what I understand, a new PR will be created everytime there is a better benchmark (using a factor 1.2, there's some room for runner differences). But doesn't this mean that the github bot will create PR seperately for package updates and benchmark updates? 

I'd rather say that we combine the update dependencies workflow with benchmark workflow (have two jobs like in release workflow for example). Then every week it's updating the packages, benchmarking and opening a PR with the benchmark results. 
If something like the scipy issue occurs, we would be able to address it on the spot. 

It's also fine if you merge this repo for now and we discuss this later.